FFmpegInterop/UncompressedVideoSampleProvider.cpp

```diff
@@ -18,140 +18,111 @@
 
 #include "pch.h"
 #include "UncompressedVideoSampleProvider.h"
-#include <mfapi.h>
 
-extern "C"
-{
-#include <libavutil/imgutils.h>
-}
-
-
-using namespace FFmpegInterop;
-
-UncompressedVideoSampleProvider::UncompressedVideoSampleProvider(
-	FFmpegReader^ reader,
-	AVFormatContext* avFormatCtx,
-	AVCodecContext* avCodecCtx)
-	: UncompressedSampleProvider(reader, avFormatCtx, avCodecCtx)
-	, m_pSwsCtx(nullptr)
-{
-	for (int i = 0; i < 4; i++)
-	{
-		m_rgVideoBufferLineSize[i] = 0;
-		m_rgVideoBufferData[i] = nullptr;
-	}
-}
+using namespace winrt::Windows::Foundation;
+using namespace winrt::Windows::Media::Core;
+using namespace winrt::Windows::Media::MediaProperties;
+using namespace winrt::Windows::Storage::Streams;
+using namespace std;
 
-HRESULT UncompressedVideoSampleProvider::AllocateResources()
+namespace winrt::FFmpegInterop::implementation
 {
-	HRESULT hr = S_OK;
-	hr = UncompressedSampleProvider::AllocateResources();
-	if (SUCCEEDED(hr))
+	UncompressedVideoSampleProvider::UncompressedVideoSampleProvider(_In_ const AVStream* stream, _In_ Reader& reader) :
+		UncompressedSampleProvider(stream, reader)
 	{
-		// Setup software scaler to convert any decoder pixel format (e.g. YUV420P) to NV12 that is supported in Windows & Windows Phone MediaElement
-		m_pSwsCtx = sws_getContext(
-			m_pAvCodecCtx->width,
-			m_pAvCodecCtx->height,
-			m_pAvCodecCtx->pix_fmt,
-			m_pAvCodecCtx->width,
-			m_pAvCodecCtx->height,
-			AV_PIX_FMT_NV12,
-			SWS_BICUBIC,
-			NULL,
-			NULL,
-			NULL);
-
-		if (m_pSwsCtx == nullptr)
+		if (m_codecContext->pix_fmt != AV_PIX_FMT_NV12)
```

Comment:
>m_codecContext->pix_fmt != AV_PIX_FMT_NV12 [](start = 6, length = 42)

The MF pipeline supports more uncompressed video formats than just NV12 - perhaps we could file a task to track expanding media type support in this sample provider in future.

Reply:
I filed a task.